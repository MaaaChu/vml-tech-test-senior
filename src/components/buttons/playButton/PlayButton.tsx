import React from "react";
import classNames from "classnames";

import styles from "./PlayButton.module.css";

import { ButtonProps } from "../../../interfaces";

interface PlayButtonProps extends ButtonProps {
  btnOnClick?: () => void;
}

const PlayButton: React.FC<PlayButtonProps> = ({
  children,
  type = "primary",
  isBtnDisabled,
  btnDisplayed = "displayed",
  btnOnClick,
}) => {
  return (
    <button
      type="button"
      disabled={isBtnDisabled}
      className={classNames(
        "btn",
        `btn--${type}`,
        `btn--${btnDisplayed}`,
        styles["btn--play"]
      )}
      onClick={btnOnClick}
    >
      {children}
    </button>
  );
};

export default PlayButton;
