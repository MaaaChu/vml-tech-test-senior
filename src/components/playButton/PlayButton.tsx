import React from "react";
import classNames from "classnames";

import styles from "./PlayButton.module.css";

import { PlayBtnDisplayed } from "../../types/Types";

interface PlayButtonProps {
  isPlaying: boolean;
  isBtnDisabled: boolean;
  btnDisplayed: PlayBtnDisplayed;
  btnOnClick: () => void;
}

const PlayButton: React.FC<PlayButtonProps> = ({
  isPlaying,
  isBtnDisabled,
  btnDisplayed,
  btnOnClick,
}) => {
  return (
    <button
      type="button"
      disabled={isBtnDisabled}
      className={classNames(
        styles["play-btn"],
        styles["play-btn--primary"],
        styles[`play-btn--${btnDisplayed}`]
      )}
      onClick={btnOnClick}
    >
      {isPlaying ? "Pause" : "Play"}
    </button>
  );
};

export default PlayButton;
