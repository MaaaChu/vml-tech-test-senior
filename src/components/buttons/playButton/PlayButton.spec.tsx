import React from "react";
import { render } from "@testing-library/react";
import PlayButton from "./PlayButton";

describe("components:PlayButton", () => {
  it("should match snapshot with string children", () => {
    const wrapper = render(
      <PlayButton type="primary" isBtnDisabled={false} btnOnClick={() => null}>
        Play
      </PlayButton>
    );

    expect(wrapper.container.firstChild).toMatchSnapshot();
  });

  it("should match snapshot when play button is disabled", () => {
    const wrapper = render(
      <PlayButton type="primary" isBtnDisabled={true} btnOnClick={() => null}>
        Play
      </PlayButton>
    );

    expect(wrapper.container.firstChild).toMatchSnapshot();
  });

  it("should match snapshot when play button is primary", () => {
    const wrapper = render(
      <PlayButton type="primary" isBtnDisabled={true} btnOnClick={() => null}>
        Play
      </PlayButton>
    );

    expect(wrapper.container.firstChild).toMatchSnapshot();
  });

  it("should match snapshot when play button is secondary", () => {
    const wrapper = render(
      <PlayButton type="secondary" isBtnDisabled={true} btnOnClick={() => null}>
        Play
      </PlayButton>
    );

    expect(wrapper.container.firstChild).toMatchSnapshot();
  });

  it("should match snapshot when play button type isnt given", () => {
    const wrapper = render(
      <PlayButton isBtnDisabled={true} btnOnClick={() => null}>
        Play
      </PlayButton>
    );

    expect(wrapper.container.firstChild).toMatchSnapshot();
  });
});
