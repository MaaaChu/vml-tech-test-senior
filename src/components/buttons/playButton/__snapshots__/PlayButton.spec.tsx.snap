// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:PlayButton should match snapshot when play button is disabled 1`] = `
<button
  class="btn btn--primary btn--displayed btn--play"
  disabled=""
  type="button"
>
  Play
</button>
`;

exports[`components:PlayButton should match snapshot when play button is primary 1`] = `
<button
  class="btn btn--primary btn--displayed btn--play"
  disabled=""
  type="button"
>
  Play
</button>
`;

exports[`components:PlayButton should match snapshot when play button is secondary 1`] = `
<button
  class="btn btn--secondary btn--displayed btn--play"
  disabled=""
  type="button"
>
  Play
</button>
`;

exports[`components:PlayButton should match snapshot when play button type isnt given 1`] = `
<button
  class="btn btn--primary btn--displayed btn--play"
  disabled=""
  type="button"
>
  Play
</button>
`;

exports[`components:PlayButton should match snapshot with string children 1`] = `
<button
  class="btn btn--primary btn--displayed btn--play"
  type="button"
>
  Play
</button>
`;
